treasury-fees-monitor/data-collection/collect-fee-data.js

```diff
@@ -0,0 +1,183 @@
+const commandLineArgs = require('command-line-args');
+const createCsvWriter = require('csv-writer').createObjectCsvWriter;
+const ethers = require('ethers');
+const fs = require('fs');
+
+require('dotenv').config();
+
+const BLOCK_CHUNK = 128;
+const SECONDS_IN_DAY = 24 * 60 * 60;
+
+async function getBlockByTimestamp(provider, userTimestamp) {
+  // seed the algorithm
+  // starting value for low block
+  let lowBlockNumber = 1;
+  let lowBlock = await provider.getBlock(lowBlockNumber);
+  let lowBlockTimestamp = lowBlock.timestamp;
+
+  // starting value for high block
+  let highBlockNumber = await provider.getBlockNumber();
+  let highBlock = await provider.getBlock(highBlockNumber);
```

Comment:
A potential alternative, given that `getBlock` will default to `latest`.
```suggestion
  let highBlock = await provider.getBlock();
  let highBlockNumber = highBlock.number;
```